source/common/http/http1/codec_impl.h

```diff
@@ -35,6 +35,16 @@ namespace Http1 {
   COUNTER(requests_rejected_with_underscores_in_headers)                                           \
   COUNTER(response_flood)
 
+// The following define special return values for http_parser callbacks. http_parser callbacks
+// should return non-zero to indicate an error and halt execution. The one exception is
+// on_headers_complete, where returning '1' will tell http_parser that it should not expect a body,
+// and returning '2' from on_headers_complete will tell http_parser that it should not expect a body
+// nor any further data on the connection.
+constexpr int HTTP_PARSER_SUCCESS = 0;
```

Comment:
why not an enum?

also, prefer HttpParserSuccess style for constants to avoid clashing with `#define` convention, which is not namespaced.

Reply:
I will namespace them to something like `HttpParserCode::Success`. I wanted to avoid `enumToInt`-ing everywhere, but it might be cleanest to change the int types to `HttpParserCode` and enumToInt-ing where the callbacks are set in `http_parser_settings`.